### PR TITLE
Update School Locations section to have "Show more" buttons on profile page

### PR DESCRIPTION
**Pull request by hughes-dustin**
## Description
https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/4383

The school locations section will have a "Show 10 more", "view all" and "view less" buttons if there are more than 10 locations. 

## Screenshots
image
##
image
##
image

## Acceptance criteria
- [x] On mobile, the initial display is the first 5 locations.
- [x] On non-mobile, the initial display is the first 10 locations.
- [x] There is an informational message stating "Showing X of Y locations" if there are more locations than specified for the display type in AC 1 & 2.
     a. X is equal to the number of locations currently displayed
     b. Y is the total number of locations that are possible to be displayed
- [x]  The user can view 10 additional locations at a time.
     a. The additional locations are appended to the list.
     b. If there are less than 10 additional locations, the button only prompts to show the remaining locations (ex. "Show 3 additional locations)
- [x] The user has the ability to view all locations at once
- [x]  Once all locations are being displayed, the user has the ability to revert the locations display the initial locations display

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by sonntag-adam — APPROVED** [review]

File: src/applications/gi/components/profile/SchoolLocations.jsx [hunk 1]
```diff
@@ -52,8 +55,30 @@ export class SchoolLocations extends React.Component {
...
+      viewableRowCount: this.state.initialRowCount,
+      viewAll: false,
+    });
+  };
+
+  showMoreClicked = () => {
+    const remainingRowCount =
+      this.state.totalRowCount - this.state.viewableRowCount;
+    if (remainingRowCount >= 10) {
+      this.setState({ viewableRowCount: this.state.viewableRowCount + 10 });
```
Comment (shawkey-daniel) [inline src/applications/gi/components/profile/SchoolLocations.jsx]: Please create a constant for the magic numbers.

File: src/applications/gi/components/profile/SchoolLocations.jsx [hunk 2]
```diff
@@ -270,23 +294,62 @@ export class SchoolLocations extends React.Component {
...
-
-    if (this.shouldHideViewMore(main, maxRows)) {
+  renderViewButtons = () => {
+    const viewableRowCount = this.state.viewableRowCount;
+    const totalRowCount = this.state.totalRowCount;
+
+    if (totalRowCount > DEFAULT_ROWS_VIEWABLE) {
+      if (viewableRowCount !== totalRowCount) {
+        const remainingRowCount = totalRowCount - viewableRowCount;
+        const showNextCount = remainingRowCount < 10 ? remainingRowCount : 10;
```
Comment (shawkey-daniel) [inline src/applications/gi/components/profile/SchoolLocations.jsx]: More magic numbers.

**Review by shawkey-daniel — CHANGES_REQUESTED** [review]

File: src/applications/gi/components/profile/SchoolLocations.jsx (hunk 1 above)
Comment (hughes-dustin) [reply src/applications/gi/components/profile/SchoolLocations.jsx]: Done!

File: src/applications/gi/components/profile/SchoolLocations.jsx (hunk 2 above)
Comment (hughes-dustin) [reply src/applications/gi/components/profile/SchoolLocations.jsx]: The magic numbers are no more.

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Review by bkjohnson — APPROVED** [review]
Unit tests would be nice here.  Maybe they could cover the acceptance criteria.

**Comment by hughes-dustin** [comment]
> Unit tests would be nice here. Maybe they could cover the acceptance criteria.

Didn't mean to ignore this request. I'll be sure to add in unit tests in a different PR.

**Pull request merged**